### PR TITLE
cpu task enqueue bug fix

### LINKED ISSUE #2246: release-1.5 When modifying guarantee gpu number, CPU resource tasks cannot enter enqueque
**What happened**:
Example，Set two queues in the cluster, A and B, 
**cluster total resource：**
```
cpu: 32
memory: 64Gi
nvidia.com/gpu: 8
```
**queue A set for**
```
  capability:
    cpu: "16"
    memory: 32Gi
    nvidia.com/gpu: "8"
  guarantee:
    resource:
      cpu: 8
      memory: 16Gi
      nvidia.com/gpu: "2"
  reclaimable: false
  weight: 1
```
**queue B set for**
```
  capability:
    cpu: "16"
    memory: 32Gi
    nvidia.com/gpu: "8"
  guarantee:
    resource:
      cpu: 8
      memory: 16Gi
      nvidia.com/gpu: "2"
  reclaimable: false
  weight: 1
```
A queues up 6 tasks, and each task resource is：
```
      cpu: 1
      memory: 1Gi
      nvidia.com/gpu: "1"
```
Tasks are running。
Change the guarantee quantity of the GPU in QUEUE B to 4。
A queue If A CPU task T1 is executed, the enqueue cannot be entered. The task resources are as follows：
```
   cpu: 1
   memory: 1Gi
```

**What you expected to happen**:
It is expected that task T1 can enter the Enqueue and be scheduled successfully

**Environment**:
- Volcano Version: release-1.5
- Kubernetes version (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.9", GitCommit:"94f372e501c973a7fa9eb40ec9ebd2fe7ca69848", GitTreeState:"clean", BuildDate:"2020-09-16T13:56:40Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.9", GitCommit:"94f372e501c973a7fa9eb40ec9ebd2fe7ca69848", GitTreeState:"clean", BuildDate:"2020-09-16T13:47:43Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from /etc/os-release):
```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```
- Kernel (e.g. `uname -a`):
`Linux cce-7tpawg0f-h8v6ksrs 3.10.0-957.12.1.el7.x86_64 #1 SMP Mon Apr 29 14:59:59 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

### PR DESCRIPTION
**Pull request by hansongChina**
#2246 Tasks whose ScalarResources are 0 or empty for minReq are not compared

### CODE REVIEW

File: pkg/scheduler/api/resource_info.go [hunk 1]
```diff
@@ -212,6 +212,29 @@ func (r *Resource) IsZero(rn v1.ResourceName) bool {
...
 }
 
+// AddDepends is used to add two given resources
+// e.g.1 r resource is <cpu 2000.00, memory 1000.00, nvidia.com/gpu 0>
+// rr resource is <cpu 3000.00, memory 2000.00 , nvidia.com/gpu 1>
+// return r resource is <cpu 5000.00, memory 3000.00, nvidia.com/gpu 0>
+// e.g.2 r resource is <cpu 2000.00, memory 1000.00>
+// rr resource is <cpu 3000.00, memory 2000.00 , nvidia.com/gpu 1>
+// return r resource is <cpu 5000.00, memory 3000.00, nvidia.com/gpu 0>
+func (r *Resource) AddDepends(rr *Resource) *Resource {
```
Comment (Thor-wl) [inline pkg/scheduler/api/resource_info.go:222]: Why not reuse function `Add`?

File: pkg/scheduler/plugins/proportion/proportion.go [hunk 2]
```diff
@@ -330,8 +330,9 @@ func (pp *proportionPlugin) OnSessionOpen(ssn *framework.Session) {
 
 		klog.V(5).Infof("job %s min resource <%s>, queue %s capability <%s> allocated <%s> inqueue <%s> elastic <%s>",
 			job.Name, minReq.String(), queue.Name, attr.realCapability.String(), attr.allocated.String(), attr.inqueue.String(), attr.elastic.String())
+		minResourceAllocated := attr.allocated.Clone().Sub(attr.elastic)
 		// The queue resource quota limit has not reached
-		inqueue := minReq.Add(attr.allocated).Add(attr.inqueue).Sub(attr.elastic).LessEqual(attr.realCapability, api.Infinity)
+		inqueue := minReq.AddDepends(minResourceAllocated).AddDepends(attr.inqueue).LessEqual(attr.realCapability, api.Infinity)
```
Comment (Thor-wl) [inline pkg/scheduler/plugins/proportion/proportion.go:335]: Why remove the caculation of `Add(attr.allocated)` and `Sub(attr.elastic)`?

Comment (hansongChina) [reply pkg/scheduler/plugins/proportion/proportion.go:335]: Without removing，MinResourceAllocated variable is (attr.allocated)Sub(attr.elastic)

File: pkg/scheduler/api/resource_info.go (hunk 1 above)
Comment (hansongChina) [reply pkg/scheduler/api/resource_info.go:222]: The function logic is to not add ScalarResources when one of its properties is 0 or empty;

**Pull request closed without merging**